core/shared/src/main/scala/matryoshka/patterns/ListF.scala

```diff
@@ -52,6 +52,14 @@ object ListF {
     case NilF()      => Seq.empty
   } (s => s.headOption.fold[ListF[A, Seq[A]]](NilF())(ConsF(_, s.tail)))
 
+  def takeUpTo[N, T, A](implicit N: Recursive.Aux[N, Option], T: Recursive.Aux[T, ListF[A, ?]]): Coalgebra[ListF[A, ?], (N, T)] =
+    pair => pair._1.project.fold[ListF[A, (N, T)]](NilF())(p => pair._2.project.map((p, _)))
+
+  def find[A](cond: A => Boolean): Algebra[ListF[A, ?], Option[A]] = {
```

Comment:
I think we could generalize this away from `ListF`, to a more basic notion of `find` for some recursive structure. Not for this PR, but I think that sort of function, might be a way to help users more easily find the correct morphism.

I wonder if that should be a second library? Or at least a different package here. Basically, base matryoshka contains the impls of the morphisms, and then the usages package wraps the direct calls to the morphisms into functions with more commonly used signatures.